backend/common/src/dense_map.rs

```diff
@@ -131,3 +131,35 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+
+    use super::*;
+
+    #[test]
+    fn len_doesnt_panic_if_lots_of_retired() {
```

Comment:
Did you mean "removed" here?

Reply:
`retired` is referring to the list of `retired` IDs in the densemap struct :)

Reply:
(I made the name less awful!)